projects/plugins/jetpack/extensions/blocks/ai-chat/question-answer.js

```diff
@@ -1,77 +1,40 @@
 /**
  * WordPress dependencies
  */
-import {
-	Button,
-	TextControl,
-	Spinner,
-	KeyboardShortcuts,
-	ExternalLink,
-} from '@wordpress/components';
+import { Button, TextControl, KeyboardShortcuts, ExternalLink } from '@wordpress/components';
 import { RawHTML, useEffect, useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
+import MarkdownIt from 'markdown-it';
 /**
  * Internal dependencies
  */
+import Loading from './loading';
 import useSubmitQuestion from './use-submit-question';
 
-const waitStrings = [
-	__( 'Good question, give me a moment to think about that 🤔', 'jetpack' ),
-	__( 'Let me work out the answer to that, back soon!', 'jetpack' ),
-	__( '🤔 Thinking, thinking, will be back with an answer soon', 'jetpack' ),
-];
+export default function QuestionAnswer() {
+	const [ answer, setAnswer ] = useState( '' );
+	const [ aiResponse, setAiResponse ] = useState( '' );
 
-// This component displays the text word by word if show animation is true
-function ShowLittleByLittle( { html, showAnimation, onAnimationDone } ) {
-	// This is the HTML to be displayed.
-	const [ displayedRawHTML, setDisplayedRawHTML ] = useState( '' );
+	const markdownConverter = new MarkdownIt( { html: true } );
 
-	useEffect(
-		() => {
-			// That will only happen once
-			if ( showAnimation && html ) {
-				// This is to animate text input. I think this will give an idea of a "better" AI.
-				// At this point this is an established pattern.
-				const tokens = html.split( ' ' );
-				for ( let i = 1; i < tokens.length; i++ ) {
-					const output = tokens.slice( 0, i ).join( ' ' );
-					setTimeout( () => setDisplayedRawHTML( output ), 50 * i );
-				}
-				setTimeout( () => {
-					setDisplayedRawHTML( html );
-					onAnimationDone();
-				}, 50 * tokens.length );
+	useEffect( () => {
+		if ( aiResponse !== '' && aiResponse !== undefined ) {
+			if ( aiResponse === '[DONE]' ) {
+				setAnswer( markdownConverter.render( `${ answer }` ) );
 			} else {
-				setDisplayedRawHTML( html );
+				setAnswer( markdownConverter.renderInline( `${ answer }${ aiResponse }` ) );
 			}
-		},
-		// eslint-disable-next-line
-		[]
-	);
-
-	return (
-		<div className="content">
-			<RawHTML>{ displayedRawHTML }</RawHTML>
-		</div>
-	);
-}
-export default function QuestionAnswer() {
-	const { question, setQuestion, answer, isLoading, submitQuestion, references } =
-		useSubmitQuestion();
+		}
+	}, [ aiResponse ] );
```

Comment:
we definitely don't want `answer` as a dependency as it will trigger an infinite rerender, so if this approach is kept we should do a functional update to  setAnswer instead to avoid this dependency warning `setAnswer(a => ...)`